File: задача2_решение.md/ISocialNetworkStrategyInstagramTwitter.py
class InstagramStrategy(SocialNetworkStrategy):
    def get_subscribers(self, user_name: str) -> List['SocialNetworkUser']:
        from ExternalLibs.Instagram import InstagramClient, InstagramUser
        
        client = InstagramClient()
        users = client.GetSubscribers(user_name)
        return [SocialNetworkUser(user.UserName) for user in users]

class TwitterStrategy(SocialNetworkStrategy):
    def get_subscribers(self, user_name: str) -> List['SocialNetworkUser']:
        from ExternalLibs.Twitter import TwitterClient, TwitterUser
        
        client = TwitterClient()
        user_id = client.GetUserIdByName(user_name)
        subscribers = client.GetSubscribers(user_id)
        return [SocialNetworkUser(client.GetUserNameById(u.UserId)) for u in subscribers]

File: задача2_решение.md/ISocialNetworkStrategyInstagramTwitter.cs
public class InstagramStrategy : ISocialNetworkStrategy
{
    public SocialNetworkUser[] GetSubscribers(string userName)
    {
        var client = new InstagramClient();
        var users = client.GetSubscribers(userName);
        return users.Select(u => new SocialNetworkUser { UserName = u.UserName }).ToArray();
    }
}

public class TwitterStrategy : ISocialNetworkStrategy
{
    public SocialNetworkUser[] GetSubscribers(string userName)
    {
        var client = new TwitterClient();
        long userId = client.GetUserIdByName(userName);
        var users = client.GetSubscribers(userId);
        return users.Select(u => new SocialNetworkUser 
                { 
                    UserName = client.GetUserNameById(u.UserId) 
                }).ToArray();
    }
}